### PR TITLE
Enabling analytical diagnosis of targets and improve calibration dashboard

### LINKED ISSUE #63: Adding a diagnostic function that checks if there exists an analytical solution to the optimization of all targets
Relaxing assumption that weights need to be positive to evaluate if targets can be hit in the first place

### LINKED ISSUE #69: Change fonts to be consistent with PolicyEngine web apps
Roboto Serif for most text, Roboto Mono for results

### LINKED ISSUE #70: New targets don't show in comparison view on the dashboard
Also can we show the table in the comparison view too?

### PR DESCRIPTION
**Pull request by juaristi22**
Fix #63 
Fix #69 
Fix #70

### CODE REVIEW

**Comment by nikhilwoodruff** [comment]
Approved on the non-analytical analysis parts but will leave @baogorek to review those.

File: src/microcalibrate/calibration.py [hunk 1]
```diff
@@ -348,6 +348,72 @@ def _assess_targets(
                         f"of records in the loss matrix. This may make calibration unstable or ineffective."
                     )
 
+    def assess_analytical_solution(
```
Comment (baogorek) [inline src/microcalibrate/calibration.py:351]: Comments about the logger:

I feel the logger is too critical here. When I use this method in an interactive environment, I get no output unless I turn on some logging options:

```
In [18]:         calibrator.assess_analytical_solution()

In [19]: import logging
    ...:
    ...: # This configures the logger to print INFO messages (and higher) to the console.
    ...: logging.basicConfig(level=logging.INFO)

In [20]:         calibrator.assess_analytical_solution()
INFO:src.microcalibrate.calibration:Assessing analytical solution to the optimization problem for each target...
This evaluates how much each target complicates achieving calibration accuracy. The loss reported is the mean squared error of the least squares solution.
INFO:src.microcalibrate.calibration:Adding: income_aged_20_30 to the above, Loss: 0.0, Change in loss: N/A
INFO:src.microcalibrate.calibration:Adding: income_aged_40_50 to the above, Loss: 0.0, Change in loss: 0.0
```
I wouldn't mind so much (at all?) if there was a data set returned with the variable name and the loss. In fact, if you did that, you could probably get rid of some logging lines.

File: src/microcalibrate/calibration.py [hunk 2]
```diff
@@ -348,6 +348,72 @@ def _assess_targets(
...
+                from scipy.sparse.linalg import lsqr
+
+                X_sparse = csr_matrix(X)
+                result = lsqr(
+                    X_sparse.T, y
+                )  # iterative method for sparse matrices
+                w_sparse = result[0]
+                y_hat = X_sparse.T @ w_sparse
+
+            return round(np.mean((y - y_hat) ** 2), 3)  # mostly for display
```
Comment (baogorek) [inline src/microcalibrate/calibration.py:390]: Not a big deal but I would suggest removing the rounding here.

File: src/microcalibrate/calibration.py [hunk 3]
```diff
@@ -348,6 +348,72 @@ def _assess_targets(
...
+        for target_name, index_list in idx_dict.items():
+            slices.append(index_list)
+            loss = _get_linear_loss(X[:, slices], y[slices], use_sparse)
+            iterative_losses.append(loss)
+            logger.info(
+                f"Adding: {target_name} to the above, Loss: {loss}, Change in loss: {iterative_losses[-1] - iterative_losses[-2] if len(iterative_losses) > 1 else 'N/A'}"
+            )
+
+        logger.setLevel(previous_level)
+
```
Comment (baogorek) [inline src/microcalibrate/calibration.py:416]: Repeating my point above: returning a data frame would be nice here.

File: src/microcalibrate/calibration.py [hunk 4]
```diff
@@ -348,6 +348,72 @@ def _assess_targets(
                         f"of records in the loss matrix. This may make calibration unstable or ineffective."
                     )
 
+    def assess_analytical_solution(
+        self, use_sparse: Optional[bool] = False
+    ) -> None:
```
Comment (baogorek) [inline src/microcalibrate/calibration.py:353]: Since this is part of the class, should we use normalization_factor?

File: tests/test_diagnostics.py [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+        weights=weights,
+        targets=targets,
+        noise_level=0.05,
+        epochs=528,
+        learning_rate=0.01,
+        dropout_rate=0,
+    )
+
+    with caplog.at_level(logging.INFO, logger="microcalibrate.calibration"):
+        calibrator.assess_analytical_solution()
```
Comment (baogorek) [inline tests/test_diagnostics.py:129]: If you return a data frame, you'll be able to check to see that the values are what you think they are. In this case they're all zero, which I think is okay for a first test, given the priority of this feature.

**Review by baogorek — CHANGES_REQUESTED** [review]
I could chat about design over a cup of coffee (for instance, should this be a separate utility or a method of Calibrate?), but really I'm interested in seeing 2 changes:

1) Please return a data frame from the analytical method function
2) In your test, use that data frame make sure that the loss is always zero. (Extra credit: use a harder target and ensure that it is not zero).

**Review by baogorek — APPROVED** [review]
Approving after trying out the most recent changes in my test environment.
